### PR TITLE
Bond dataset generation (wip)

### PR DESCRIPTION
**Pull request by danieljcabrera**
## Description
Added a pipeline that filters for single nitrogen-nitrogen bonds.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [ ] Create more unique fingerprint functions

## Questions
- [ ] 

## Status
- [ ] Ready to go

### CODE REVIEW

**Comment by btjanaka** [comment]
@jmaat  @danieljcabrera Would you like feedback on this PR?

**Comment by danieljcabrera** [comment]
Yes, any feedback would be much appreciated! 

>

File: n-n_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+'''
```
Comment (btjanaka) [inline n-n_pipeline.py:1]: Python filenames typically use snakecase, e.g. `this_is_a_file.py`. I think a name like `nn_pipeline.py` or `n_n_pipeline.py` would be better.

File: n-n_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Script for filtering a dataset for molecules with single 
+nitrogen-nitrogen bonds
+Usage:
+    python n-n_pipeline.py --smiles-database database.smi
```
Comment (btjanaka) [inline n-n_pipeline.py:5]: See filename comment.

File: n-n_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        "dataset_type": "SMILES",
+        "dataset_info": "dataset.smi",
+        "in_memory_sorting_threshold": 25000,
+    }
+    
+    run_dance(config)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (btjanaka) [inline n-n_pipeline.py:45]: GitHub currently shows there's no newline at the end of the file.

File: n-n_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--smiles-database",
+                        type=str,
+                        required=True,
+                        help=("Name of a SMILES file containing molecules "
+                              "to process in the pipeline."))
+    args = parser.parse_args()
+
+    def relevant_if_has_single_bonded_nitrogen(mol):
+        subs = oechem.OESubSearch("[#7:1]-[#7:2]")
```
Comment (btjanaka) [inline n-n_pipeline.py:24]: Nice :D

**Review by btjanaka — COMMENTED** [review]
Some thoughts:

- The code itself looks good, and it looks like there's more to come :)
- I think you should put this file in the `examples/` directory -- the top-level directory in a Python project is usually for things like config files and READMEs.
- I see that you are uploading files to git using the web UI, as some of the commits have messages like "Add files via upload". There should be a box for adding more detailed messages when you upload. In general, though, the command line interface to git is much more flexible, and I think you should use that instead (let me know if you need any pointers to learning about git)
- It looks like the commits mainly apply to one file. When you merge the PR, consider doing a "merge and squash", which combines all the commits into one commit before putting them on the `master` branch, instead of putting on all the individual commits.
- It seems the Travis CI build is failing. I don't think it's important for this commit, but it will be important if you are going to develop the actual `dance` package. I'll look into it over the weekend. I'll also add you as a contributor to the repo at that time.

**Comment by btjanaka** [comment]
Closing this -- see #11

**Pull request closed without merging**